back-end/src/modules/auth/UserApiKeyController.ts

```diff
@@ -0,0 +1,35 @@
+import { ContentType, Controller, Get, PathParams, Head, Returns, HeaderParams} from '@tsed/common';
+import { MongooseModel } from '@tsed/mongoose';
+import { Inject } from '@tsed/di';
+import { UserApiKey, authenticate } from './UserApiKey';
+import { User } from './user/User';
+import { AuthenticationService } from './services/AuthenticationService';
+
+@Controller('')
+export class UserApiKeyController {
+  @ContentType('application/json')
+  @Inject(UserApiKey)
+  private userApiKeyModel: MongooseModel<UserApiKey>;
+  @Inject(AuthenticationService)
+  private authenticationService = new AuthenticationService
+  
+
+  @Get('/login')
+  @Returns(UserApiKey)
+  async getApiKey(@HeaderParams() headers: authenticate) : Promise<any> {
+      const apiKey = await this.authenticationService.authenticate(headers);
+      console.log(apiKey.user)
+      return {
+        token: apiKey.token,
+        validUntil: apiKey.validUntil
+      }
+  }
+
+  @Get('/sessie/:id')
```

Comment:
I would make this /session

Reply:
And then receive this based on the authorization header information 😊 (the bearer token) 

Instead of getting it by the :id